float32[] value
int32     type
int32     pass
ExJoints  exjoints

